recipes/args/all/conanfile.py

```diff
@@ -0,0 +1,108 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class ArgsConan(ConanFile):
+    name = "args"
+    description = "Small open-source library for program argument parser, inspired by Python's `argparse`, " \
+                  "depending only on the standard library, with C++17 as minimum requirement."
+    homepage = "https://github.com/mbits-libs/args"
+    license = "MIT"
+    topics = ("command-line", "commandline", "commandline-interface",
+              "program-arguments", "argparse", "argparser", "argument-parsing")
+
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        compiler = str(self.settings.compiler)
+        if compiler in self._working_compilers:
+            working_compiler = self._working_compilers[compiler]
+            current_compiler = ArgsConan.Version(
+                str(self.settings.compiler.version))
+            if working_compiler > current_compiler:
```

Comment:
We have been using [this structure](https://github.com/conan-io/conan-center-index/blob/8339c676c031475406dad9b5749f58eeca60eab0/recipes/magic_enum/all/conanfile.py#L38-L46) for the minimum compiler version, it would be nice to have something more consistent, I personally havent seen this in any recipe.

Reply:
I haven't found anything about minimal compiler _until_ I started looking into usage of `cppstd` (poor search-foo on my side). Updated.

Reply:
No worries, the docs are will a WIP. thanks!